examples/chat-room-history/javascript/src/script.ts

```diff
@@ -0,0 +1,102 @@
+import * as Ably from 'ably';
+import { ChatClient, Message, RoomOptionsDefaults } from '@ably/chat';
+import { faker } from '@faker-js/faker';
+import './styles.css';
+
+const realtimeClient = new Ably.Realtime({
+  clientId: faker.person.firstName(),
+  key: import.meta.env.VITE_PUBLIC_ABLY_KEY as string,
+});
+const chatClient = new ChatClient(realtimeClient);
+const room = chatClient.rooms.get('chat-room-messages', RoomOptionsDefaults);
+
+const landingPage = document.getElementById('landing-page');
+const chatRoom = document.getElementById('chat-room-messages');
+
+/** Action to enter the chat room, loading in historical messages, and subscribing to any new messages received. */
+document.getElementById('enter-chat').addEventListener('click', async () => {
+  landingPage.style.display = 'none';
+  chatRoom.style.display = 'block';
+  await enterChat();
+});
+
+/** Action to load a random message into the chat room. To be used pre-loading historical messages before entering. */
+document.getElementById('load-message').addEventListener('click', () => {
+  const randomMessage = generateRandomMessage();
+  room.messages.send({ text: randomMessage });
+
+  // Create and show alert
+  const alert = document.createElement('div');
+  alert.className = 'fixed top-4 right-4 bg-green-500 text-white px-6 py-3 rounded shadow-lg';
+  alert.textContent = `Message: "${randomMessage}" sent to chat room!`;
+  document.body.appendChild(alert);
+
+  // Remove alert after 3 seconds
+  setTimeout(() => {
+    alert.remove();
+  }, 3000);
+});
+
+function generateRandomMessage() {
+  const messages = [
+    'Hey everyone! Just joined this amazing chat.',
+    'What a beautiful day for chatting!',
+    'Hello from the other side of the screen!',
+    'Anyone here interested in web development?',
+    'I love how this chat app works!',
+    'Greetings from a random message generator!',
+  ];
+  return messages[Math.floor(Math.random() * messages.length)];
+}
+
+async function enterChat() {
+  /** 💡 Add every every message published to the room 💡 */
+  const { getPreviousMessages } = room.messages.subscribe((message) => {
+    addMessage(message.message);
+
+    const messagesContainer = document.getElementById('messages');
+    messagesContainer.scrollTop = messagesContainer.scrollHeight;
+  });
+
+  /** 💡 Attach to the room to subscribe to messages 💡 */
+  await room.attach();
+
+  const historicalMessages = await getPreviousMessages({ limit: 10 });
+  historicalMessages.items.reverse().forEach((message) => {
```

Comment:
You'll be pleased to know we are adding an `orderBy` param which will take `newestFirst` and `oldestFirst` to clean up any confusion in this area! :D

Reply:
Do you know when we can expect this to be out in the wild?

Reply:
Also next week! :D As part of the v0.3.0 release